pkg/agent/endpoints/sdsv3/handler.go

```diff
@@ -307,6 +337,171 @@ func (h *Handler) triggerReceivedHook() {
 	}
 }
 
+type validationContextBuilder interface {
+	buildOne(resourceName, trustDomainID string) (*any.Any, error)
+	buildAll(resourceName string) (*any.Any, error)
+}
+
+type rootCABuilder struct {
+	bundles map[string]*bundleutil.Bundle
+}
+
+func newRootCABuilder(bundle *bundleutil.Bundle, federatedBundles map[spiffeid.TrustDomain]*bundleutil.Bundle) validationContextBuilder {
+	bundles := make(map[string]*bundleutil.Bundle, len(federatedBundles)+1)
+	// Add td bundle if provided
```

Comment:
FWIW, tdBundle should never be nil. I'm ok with the defensive code, but we should adjust the comment so folks don't think it can be nil in normal circumstances.
```suggestion
	// Only include tdBundle if it is not nil, which shouldn't ever be the case. This is purely defensive.
```